### PR TITLE
Decaff all the things Dredd, round two

### PR DESCRIPTION
**Pull request by michalholasek**
Here goes second PR from the decaffenaite series.

### Notable changes:
- Files in `test`directory `.coffee` -> `.js`
- `test` files are also built to `dist` directory for testing / coverage purposes
- `script/coverage.sh` was slightly modified again

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,21 +7,26 @@
     "node": ">= 4"
   },
   "scripts": {
-    "build": "babel src --out-dir ./dist/src",
+    "build:source": "babel src --out-dir ./dist/src",
+    "build:test": "babel test --out-dir ./dist/test --copy-files",
     "clean": "rimraf dist",
-    "copy:test": "node scripts/copy.js",
     "coveralls": "scripts/coveralls.sh",
+    "make": "npm run build:source && npm run build:test",
```
Comment (honzajavorek) [inline package.json]: Wouldn't `build` make more sense then `make`?

File: scripts/coverage.sh [hunk 2]
```diff
@@ -9,21 +9,17 @@ PROJECT_DIR=$(pwd -P)
...
+npm run make
 
 
 # Testing
 export COVERAGE_FILE
-istanbul cover ./node_modules/.bin/mocha -- \
-    "$TEST_GLOB" \
-    --compilers=coffee:coffee-script/register \
-    --recursive
+istanbul cover ./node_modules/.bin/_mocha -- "$TEST_GLOB" --opts=./mocha.opts
```
Comment (honzajavorek) [inline scripts/coverage.sh]: In npm scripts it seems using just `dist/test` is sufficient and no glob is needed.

Comment (honzajavorek) [reply scripts/coverage.sh]: Didn't we want to move the opts back to the test directory? Then we could omit the `--opts` option here and in package.json.

File: package.json [hunk 3]
```diff
@@ -7,21 +7,26 @@
     "node": ">= 4"
   },
   "scripts": {
-    "build": "babel src --out-dir ./dist/src",
+    "build:source": "babel src --out-dir ./dist/src",
+    "build:test": "babel test --out-dir ./dist/test --copy-files",
```
Comment (honzajavorek) [inline package.json]: nit: I usually prefer gnu style options `--out-dir=./dist/test`

File: scripts/coverage.sh (hunk 2 above)
Comment (honzajavorek) [reply scripts/coverage.sh]: Hmm... Correct me if I'm wrong, but this file could be now completely rewritten as an npm script:

```
"coverage": "rm -rf ./coverage && npm run build && istanbul cover _mocha -- dist/test"
```

And I'm not completely sure whether `rm -rf ./coverage` is necessary.

File: package.json [hunk 4]
```diff
@@ -7,15 +7,15 @@
...
     "clean": "rimraf dist",
     "coveralls": "scripts/coveralls.sh",
-    "make": "npm run build:source && npm run build:test",
+    "build": "npm run build:source && npm run build:test",
     "lint": "scripts/commitlint.sh && eslint src",
     "prepublish": "npm run build:source",
     "semantic-release": "semantic-release pre && npm publish && semantic-release post",
-    "test": "npm run clean && npm run make && mocha dist/test --opts ./mocha.opts",
+    "test": "npm run clean && npm run build && mocha dist/test --opts=./mocha.opts",
     "test:coverage": "scripts/coverage.sh"
```
Comment (honzajavorek) [inline package.json]: This contains `npm run make` as well, that's why Travis fails.

File: package.json (hunk 3 above)
Comment (michalholasek) [reply package.json]: Does not work for `mocha`'s `--opts=./mocha.opts` unfortunately. I'll leave it as it is for consistency.

File: package.json (hunk 1 above)
Comment (michalholasek) [reply package.json]: Addressed in https://github.com/apiaryio/dredd-transactions/pull/128/commits/606f57c30ae991d3f20359b3d4ba35d1a3d13c00.

File: scripts/coverage.sh (hunk 2 above)
Comment (michalholasek) [reply scripts/coverage.sh]: Addressed in https://github.com/apiaryio/dredd-transactions/pull/128/commits/e5f0e92e37a9f86c4d73d08d58fa5cf197387f70 and https://github.com/apiaryio/dredd-transactions/pull/128/commits/1755b818097d45525c052e264a420a3782bfeb96.

File: package.json (hunk 3 above)
Comment (honzajavorek) [reply package.json]: Oh mocha 🤦‍♂️  OK

Comment (honzajavorek) [reply package.json]: 😏 the last commits don't contain --opts anymore 😄 but let it be

**Pull request merged**